src/application/predictor.hpp

```diff
@@ -85,6 +85,10 @@ class Predictor {
         }
       };
     } else if (predict_contrib) {
+      if (boosting_->IsLinear()) {
+        Log::Warning("Predicting SHAP feature contributions is not implemented for linear trees. "
```

Comment:
Maybe raising error will be better? I believe it is better than returning nonsense results.
```suggestion
        Log::Fatal("Predicting SHAP feature contributions is not implemented for linear trees. "
```